course4/homework-linked-list/count_until.hh

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <iterator>
+
+namespace course {
+
+template<class InputIt, class T>
+typename std::iterator_traits<InputIt>::difference_type
+    count_until(InputIt first, InputIt last, const T& value, const T& stop)
+{
+    typename std::iterator_traits<InputIt>::difference_type ret = 0;
+    for (; (first != last) && (*first)!=stop; ++first) {

```

Comment:
The first solution looks very nice,m although, indeed, int will iterate over the collection twice.

Why do you think that the second one would be worse?